sphinx/environment/__init__.py

```diff
@@ -677,15 +692,37 @@ def read_doc(self, docname, app=None):
         language = self.config.language or 'en'
         self.settings['language_code'] = language
         if 'smart_quotes' not in self.settings:
-            self.settings['smart_quotes'] = True
+            self.settings['smart_quotes'] = self.config.smart_quotes
             if self.config.html_use_smartypants is not None:
                 warnings.warn("html_use_smartypants option is deprecated. Smart "
                               "quotes are on by default; if you want to disable "
-                              "or customize them, use the smart_quotes option in "
-                              "docutils.conf.",
+                              "them, use the smart_quotes option.",
                               RemovedInSphinx17Warning)
                 self.settings['smart_quotes'] = self.config.html_use_smartypants
 
+            # some conditions exclude smart quotes, overriding smart_quotes
+            for valname, vallist in iteritems(self.config.smartquotes_excludes):
+                if valname == 'builders':
+                    # this will work only for checking first build target
+                    if self.app.builder.name in vallist:
+                        self.settings['smart_quotes'] = False
+                        break
+                elif valname == 'languages':
+                    if self.config.language in vallist:
+                        self.settings['smart_quotes'] = False
+                        break
+                else:
+                    try:
+                        attr = getattr(self.config, valname[:-1])
+                        if attr in vallist:
+                            self.settings['smart_quotes'] = False
+                            break
+                    except AttributeError:
+                        warnings.warn("'%s' (from '%s' in smartquotes_excludes) "
+                                      "isn't a valid configuration variable. "
+                                      "Did you forget an ending 's'?"
+                                      % (valname[:-1], valname))
```

Comment:
perhaps the whole ``else`` clause could be dropped if we limit to conditioning on language and builder

Reply:
As I commented above, I can't imagine the use case for this else block.
I don't opposite it, but I'd like to know an example use case before merging.

Reply:
sorry I overlooked your comment, I will remove that block as I don't see obvious use case either

Reply:
+1